### PR TITLE
[MAINT] Format datasets tests func

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates https://github.com/nilearn/nilearn/issues/2528

Splinter of https://github.com/nilearn/nilearn/pull/3624

Changes proposed in this pull request:

- apply isort, black and precommit hooks
- fix flake8 issues

will update CI checks, pre-commit and pyproject.toml only in the last PR for the dataset folder.

### CODE REVIEW

**Review by ymzayek — APPROVED** [review]
LGTM!

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: nilearn/datasets/tests/test_func.py [hunk 1]
```diff
@@ -32,47 +29,55 @@ def _load_localizer_index():
...
+        )
+    localizer_index["/localizer/phenotype/behavioural.tsv"] = uuid.uuid4().hex
+    localizer_index["/localizer/participants.tsv"] = uuid.uuid4().hex
     tsv_files = {}
-    tsv_files['/localizer/phenotype/behavioural.tsv'] = pd.read_csv(
-        str(data_dir / 'localizer_behavioural.tsv'), sep='\t')
-    tsv_files['/localizer/participants.tsv'] = pd.read_csv(
-        str(data_dir / 'localizer_participants.tsv'), sep='\t')
+    tsv_files["/localizer/phenotype/behavioural.tsv"] = pd.read_csv(
+        str(data_dir / "localizer_behavioural.tsv"), sep="\t"
```
Comment (NicolasGensollen) [inline nilearn/datasets/tests/test_func.py:40]: We don't need to cast to string here:

```suggestion
        data_dir / "localizer_behavioural.tsv", sep="\t"
```

File: nilearn/datasets/tests/test_func.py [hunk 2]
```diff
@@ -32,47 +29,55 @@ def _load_localizer_index():
...
+    )
+    tsv_files["/localizer/participants.tsv"] = pd.read_csv(
+        str(data_dir / "localizer_participants.tsv"), sep="\t"
```
Comment (NicolasGensollen) [inline nilearn/datasets/tests/test_func.py:43]:
```suggestion
        data_dir / "localizer_participants.tsv", sep="\t"
```

File: nilearn/datasets/tests/test_func.py [hunk 3]
```diff
@@ -350,184 +408,156 @@ def test_check_parameters_megatrawls_datasets(request_mocker):
...
 
 
 def test_fetch_megatrawls_netmats(tmp_path, request_mocker):
     # smoke test to see that files are fetched and read properly
     # since we are loading data present in it
-    files_dir = str(tmp_path / 'Megatrawls'
-                    / '3T_Q1-Q6related468_MSMsulc_d100_ts3')
+    files_dir = str(
+        tmp_path / "Megatrawls" / "3T_Q1-Q6related468_MSMsulc_d100_ts3"
+    )
```
Comment (NicolasGensollen) [inline nilearn/datasets/tests/test_func.py:435]: unrelated to formatting, but we don't need to use `os.makedirs` or `os.path.join`, we can rely on Pathlib  (`files_dir.mkdir(parents=True)`, `files_dir / "Znet2.txt"`....)

File: nilearn/datasets/tests/test_func.py [hunk 4]
```diff
@@ -641,74 +680,75 @@ def test_fetch_bids_langloc_dataset(request_mocker, tmp_path):
...
+    urls = [
+        data_prefix + "/stuff.html",
+        data_prefix + "/sub-xxx.html",
+        data_prefix + "/sub-yyy.html",
+        data_prefix + "/sub-xxx/ses-01_task-rest.txt",
+        data_prefix + "/sub-xxx/ses-01_task-other.txt",
+        data_prefix + "/sub-xxx/ses-02_task-rest.txt",
+        data_prefix + "/sub-xxx/ses-02_task-other.txt",
+        data_prefix + "/sub-yyy/ses-01.txt",
+        data_prefix + "/sub-yyy/ses-02.txt",
```
Comment (NicolasGensollen) [inline nilearn/datasets/tests/test_func.py:697]: We can use a list comprehension here to avoid copy pasting data_prefix.

**Review by NicolasGensollen — APPROVED** [review]
Thanks @Remi-Gau !
The re-formatting LGTM.
I made a few unrelated comments about Pathlib for future PRs.

**Comment by Remi-Gau** [comment]
> Thanks @Remi-Gau ! The re-formatting LGTM. I made a few unrelated comments about Pathlib for future PRs.

Thanks for the hints

will put add that list of refactoring suggestions to a card on this [project](https://github.com/nilearn/nilearn/projects/8) to make sure I address them in another PR.

**Pull request merged**